### PR TITLE
[test] Update configuration and tests for MCH system

### PR DESCRIPTION
**Pull request by kraushm**
Changes to configuration and tests to include new PEs on Tsa/Arolla.

### CODE REVIEW

**Comment by kraushm** [comment]
@jenkins-cscs retry all

File: config/cscs.py [hunk 1]
```diff
@@ -781,6 +805,18 @@
             'cxx': 'g++',
             'ftn': 'gfortran'
         },
+        {
+            'name': 'PrgEnv-pgi-nompi-nocuda',
+            'target_systems': [
+                'arolla'
```
Comment (vkarak) [inline config/cscs.py:811]:
```suggestion
                'arolla', 'tsa'
```

And remove the same entry for Tsa below.

File: config/cscs.py [hunk 2]
```diff
@@ -805,6 +841,18 @@
             'cxx': 'mpicxx',
             'ftn': 'mpifort'
         },
+        {
+            'name': 'PrgEnv-pgi-nocuda',
+            'target_systems': [
+                'arolla'
```
Comment (vkarak) [inline config/cscs.py:847]: Similarly here.

File: config/cscs.py [hunk 3]
```diff
@@ -817,6 +865,18 @@
             'cxx': 'mpicxx',
             'ftn': 'mpifort'
         },
+        {
+            'name': 'PrgEnv-gnu-nocuda',
+            'target_systems': [
+                'arolla'
```
Comment (vkarak) [inline config/cscs.py:871]: Similarly here.

File: config/cscs.py [hunk 4]
```diff
@@ -829,6 +889,18 @@
             'cxx': 'g++',
             'ftn': 'gfortran'
         },
+        {
+            'name': 'PrgEnv-gnu-nompi-nocuda',
+            'target_systems': [
+                'arolla'
```
Comment (vkarak) [inline config/cscs.py:895]: Similarly here.

File: cscs-checks/prgenv/helloworld.py [hunk 5]
```diff
@@ -89,7 +90,12 @@ def num_ranks(match):
 
     @rfm.run_before('compile')
     def setflags(self):
-        envname = self.current_environ.name.replace('-nompi', '')
+        if '-nompi-nocuda' in self.current_environ.name:
+            envname = self.current_environ.name.replace('-nompi-nocuda', '')
+        elif '-nocuda' in self.current_environ.name:
+            envname = self.current_environ.name.replace('-nocuda', '')
+        else:
+            envname = self.current_environ.name.replace('-nompi', '')
```
Comment (vkarak) [inline cscs-checks/prgenv/helloworld.py:98]: Replace all this with the following one liner:

```suggestion
        envname = re.sub(r'(PrgEnv-\w+).*', lambda m: m.group(1), self.current_environ.name)
```

**Comment by victorusu** [comment]
@kraushm, I agree with @vkarak's comments.

**Comment by vkarak** [comment]
@jenkins-cscs retry daint

**Pull request merged**